api/app/controllers/mno_enterprise/jpi/v1/admin/invoices_controller.rb

```diff
@@ -54,6 +64,45 @@ def last_commission_amount
 
     private
 
+    def invoice_adjustments
+      bills = @invoice.bills
+      params[:invoice][:adjustments].each do |adjustment|
+        attributes = adjustment.permit(*ADJUSTMENT_ATTRIBUTES).merge(bill_attributes)
```

Comment:
Hello, I think you could have the same approach that we did for the Product and Product Subscription

```
    if pricing_params
      @product = @product.load_required(:product_pricings)
      id_to_pricing = @product.product_pricings.map { |p| [p.id, p] }.to_h
      pricing_params.each do |p|
        attributes = p.permit(*PRICING_ATTRIBUTES)
        pricing = id_to_pricing.delete(p[:id])
        if pricing
          pricing.update(attributes)
        else
          pricing = MnoEnterprise::ProductPricing.create(attributes.merge(product_id: @product.id))
        end
        return render json: pricing.errors, status: :bad_request if pricing.errors.any?
      end
      id_to_pricing.each_value { |p| p.destroy }
    end`
```
which has the advantage to not reload the bills